troop/views.py

```diff
@@ -280,3 +283,65 @@ def participant_row(fields, attendance_days, participant):
                 yield 1 if field in days else ""
             else:
                 yield getattr(participant, field)
+
+
+class SendEmailView(OnlyTroopManagerMixin, generic.FormView):
+    template_name = "troop/email_form.html"
+    form_class = SendEmailForm
+
+    def get_form_kwargs(self, **kwargs):
+        form_kwargs = super().get_form_kwargs(**kwargs)
+        form_kwargs["user"] = self.request.user
+        form_kwargs["troop"] = self.request.troop
+        return form_kwargs
+
+    def get_success_url(self):
+        kwargs = {"troop_number": self.request.troop.number}
+        return reverse("troop:participant.index", kwargs=kwargs)
+
+    def form_valid(self, form):
+        sender = form.cleaned_data["sender"]
+        reply_to = form.cleaned_data["reply_to"]
+        section = form.cleaned_data["section"]
+        status = form.cleaned_data["status"]
+        subject = form.cleaned_data["subject"]
+        message = form.cleaned_data["message"]
+
+        participants = self.request.troop.participant_set.exclude(email="")
+
+        if section != "all" and section != "no":
+            participants = participants.filter(age_section=section)
+        elif section == "no":
+            participants = participants.filter(age_section="")
+
+        if status == "lea":
+            participants = participants.filter(is_leader=True)
+
+        if status == "chi":
+            participants = participants.filter(is_leader=False)
+
+        headers = {"Reply-To": reply_to}
+
+        emails = [
+            EmailMessage(subject, message, sender, [p.email], headers=headers,)
+            for p in participants
+        ]
+
+        # Apparently, send_mass_mail() only takes tuples so this is not possible:
+        # send_mass_mail(emails, fail_silently=False)
```

Comment:
We should use https://docs.djangoproject.com/en/3.0/topics/email/#sending-multiple-emails to prevent opening many SMTP sessions.

Reply:
Thanks, somehow missed this in the docs. I fixed it.